src/Network/CurlResult.php

```diff
@@ -0,0 +1,296 @@
+<?php
+
+namespace Friendica\Network;
+
+
+use Friendica\Network\HTTPException\InternalServerErrorException;
+
+/**
+ * A content class for Curl call results
+ */
+class CurlResult
+{
+	/**
+	 * @var int HTTP return code or 0 if timeout or failure
+	 */
+	private $returnCode;
+
+	/**
+	 * @var string the content type of the Curl call
+	 */
+	private $contentType;
+
+	/**
+	 * @var string the HTTP headers of the Curl call
+	 */
+	private $header;
+
+	/**
+	 * @var boolean true (if HTTP 2xx result) or false
+	 */
+	private $isSuccess;
+
+	/**
+	 * @var string the URL which was called
+	 */
+	private $url;
+
+	/**
+	 * @var string in case of redirect, content was finally retrieved from this URL
+	 */
+	private $redirectUrl;
+
+	/**
+	 * @var string fetched content
+	 */
+	private $body;
+
+	/**
+	 * @var array some informations about the fetched data
+	 */
+	private $info;
+
+	/**
+	 * @var boolean true if the URL has a redirect
+	 */
+	private $isRedirectUrl;
+
+	/**
+	 * @var boolean true if the curl request timed out
+	 */
+	private $isTimeout;
+
+	/**
+	 * @var int the error number or 0 (zero) if no error
+	 */
+	private $errorNumber;
+
+	/**
+	 * @var string the error message or '' (the empty string) if no
+	 */
+	private $error;
+
+	/**
+	 * Creates an errored CURL response
+	 *
+	 * @param string $url optional URL
+	 *
+	 * @return CurlResult a CURL with error response
+	 */
+	public static function createErrorCurl($url = '')
+	{
+		return new CurlResult($url,'', [ 'http_code' => 0 ]);
```

Comment:
Code standards: Please add a space after commas and remove spaces inside square brackets.